src/NetTopologySuite.IO.GeoJSON/Converters/GeometryConverter.cs

```diff
@@ -407,8 +407,15 @@ private Geometry ParseGeometry(JsonReader reader, JsonSerializer serializer)
                         if (geometryType == null)
                         {
                             reader.ReadOrThrow();
-                            geometryType = (GeoJsonObjectType)Enum.Parse(typeof(GeoJsonObjectType),
-                                (string)reader.Value, true);
+                            try
+                            {
+                                geometryType = (GeoJsonObjectType)Enum.Parse(typeof(GeoJsonObjectType),
+                                    (string)reader.Value, true);
+                            }
+                            catch (ArgumentException)
+                            {
+                                throw new ParseException($"Unknown geometry type '{(string)reader.Value}'");
+                            }
```

Comment:
(minor) slightly nicer:
```suggestion
                            geometryType = Enum.TryParse((string)reader.Value, true, out GeoJsonObjectType parsedGeometryType)
                                ? parsedGeometryType
                                : throw new ParseException($"Unknown geometry type '{(string)reader.Value}'");
```